staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -916,8 +938,7 @@ func (c *Cacher) startDispatchingBookmarkEvents() {
 				continue
 			}
 			c.watchersBuffer = append(c.watchersBuffer, watcher)
-			// Requeue the watcher for the next bookmark if needed.
-			c.bookmarkWatchers.addWatcher(watcher)
+			c.expiredBookmarkWatchers = append(c.expiredBookmarkWatchers, watcher)
```

Comment:
Trying to minimize the core PR again. This is effectively a refactoring that we can do independently.
Can you maybe move the no-op changes to a separate PR:
- the introduction of expiredBookmarkWatchers
- rename of function XxxLocked
- most of comment updated in: nextBookmarkTime [eventually let's add your new condition as (c) and let's make that incremental over the no-op PR factored out of this one]
-

Reply:
sure, np

Reply:
done https://github.com/kubernetes/kubernetes/pull/111244